snuba/subscriptions/codecs.py

```diff
@@ -62,3 +69,44 @@ def encode(self, value: SubscriptionTaskResult) -> KafkaPayload:
             ).encode("utf-8"),
             [],
         )
+
+
+class SubscriptionScheduledTaskEncoder(Encoder[bytes, ScheduledTask[Subscription]]):
+    """
+    Encodes/decodes a scheduled subscription to bytes for Kafka.
+    Does not support non SnQL subscriptions.
+    """
+
+    def __init__(self, entity_key: EntityKey) -> None:
+        self.__entity_key = entity_key
+
+    def encode(self, value: ScheduledTask[Subscription]) -> bytes:
+        assert isinstance(value.task.data, SnQLSubscriptionData)
+
+        return cast(
+            bytes,
+            rapidjson.dumps(
+                {
+                    "timestamp": value.timestamp.isoformat(),
+                    "task": {
+                        "identifier": str(value.task.identifier),
+                        "data": value.task.data.to_dict(),
```

Comment:
I wonder if we even need to bother serializing the task data instead of just providing the subscription id and the timestamp and let the executor figure out the storage.  Let's keep an eye on the impact of this when you will profile the scheduler.